### DISCUSSION TITLE
cannot load audio file from cloud front distribution (AWS)

### DISCUSSION THREAD

**Discussion by hieunm22** [katspaugh/wavesurfer.js#3956] [Q&A] [answered]
I tried to use a signed url using cloud front distribution which restrict access to an S3 bucket
quick brief about my architecture: there's a private S3 bucket which restricts access to it. all accesses to objects in this bucket need to access to a cloudfront distribution with a pre-signed url

![image](https://github.com/user-attachments/assets/4ee16b28-1871-44ad-bb1b-a677b4d81215)

when loading the wavesufer to UI and assign source url to a pre-sign url like this, it shows an error as image above
https://<distribution_name>.cloudfront.net/<file_key>?Expires=1733748078&Signature=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&Key-Pair-Id=KXXXXXXXXXXXXX

(this url works fine in browser, runs correctly)

Note: this works when using a SAS token in azure storage

how to fix it to play audio from cloudfront distribution?
I don't know is it a bug or not so I post a question here.
Thanks

**Comment by katspaugh** [comment]
Try searching in the forum and in the issues, there were some tips people posted on how to set up S3 for CORS.

**Comment by hieunm22** [comment, marked as answer]
find my solution by editing s3 cors configuration in console.
thanks
